app/javascript/components/forms/firmware-update.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from "react";
+import { Form, Field, FormSpy } from "react-final-form";
+import { Form as PfForm, Grid, Button, Col, Row, Spinner } from "patternfly-react";
+import PropTypes from "prop-types";
+import { required } from "redux-form-validators";
+
+import { FinalFormField, FinalFormSelect } from "@manageiq/react-ui-components/dist/forms";
+import '@manageiq/react-ui-components/dist/forms.css';
+
+const FirmwareUpdate = ({loading, updateFormState, physicalServerIds, initialValues, firmwareBinaries}) => {
+  if(loading){
+    return (
+      <Spinner loading size="lg" />
+    );
+  }
+
+  return (
+    <Form
+      onSubmit={() => {}} // handled by modal
+      initialValues={initialValues}
+      render={({ handleSubmit }) => (
+        <PfForm horizontal>
+          <FormSpy onChange={state => updateFormState({ ...state, values: state.values })} />
+          <Grid fluid>
+            <Row>
+              <Col xs={12}>
+                <h2>{__(`Number of servers to update firmware for: ${physicalServerIds.length}`)}</h2>
```

Comment:
This will not be translated, you need to use `sprintf` to assign variables to getText strings

```javascript
sprintf(__('Number of servers to update firmware for: %s.'), physicalServerIds.length)
```

You also have to create a mock of sprintf for your tests.

Reply:
Thank you. Instead mocking sprintf I just imported the thing in my setup.js or else snapshots would include %s instead interpolated value.